### PR TITLE
[YANG] Add valid v6 address to sonic-dhcp-sever.yang

### PR DESCRIPTION
**Pull request by wen587**
#### Why I did it
The PR is to allow v6 address in DHCP_SERVER table which was generated from PROD minigraph. This is to pass the YANG validation.
##### Work item tracking
- Microsoft ADO **(number only)**: 27735460

#### How I did it
Replace ipv4-address with ip-address which is a union of ipv4-address nad ipv6-address
#### How to verify it
Unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
         },
         "DHCP_SERVER": {
             "192.0.0.8": {},
-            "192.0.0.8": {}
```
Comment (qiluo-msft) [inline src/sonic-yang-models/tests/files/sample_config_db.json:10]: Do not delete old test data. I believe it is intentionally duplicating above item, and it is a corner case to test.

Comment (wen587) [reply src/sonic-yang-models/tests/files/sample_config_db.json:10]: This is added in sonic-dhcp_server.yang PR: https://github.com/sonic-net/sonic-buildimage/pull/12175
It should not be related to corner case since the build has passed.

**Review by Blueve — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by yaqiangz — APPROVED** [review]

**Comment by yaqiangz** [comment]
@yxieca could you please help to approve this backport to 202311?
